client/electron/src/updater.ts

```diff
@@ -47,10 +48,29 @@ class CustomGithubProvider extends GitHubProvider {
   }
 }
 
-const publishOption: CustomPublishOptions | CustomGitHubOptions = {
-  ...require('../assets/publishConfig.json'),
-  updateProvider: CustomGithubProvider,
-};
+function loadPublishOption(): (CustomPublishOptions & CustomGitHubOptions) | undefined {
+  let rawData = undefined;
+  try {
+    rawData = readFileSync('../assets/publishConfig.json', 'utf8');
+  } catch {
+    console.log('Failed to load publish config file');
+    return undefined;
+  }
+  const data = JSON.parse(rawData);
```

Comment:
Can't you use import for that? `loadPublishOption` would need to be async but you should be able to do
```typescript
try {
  config = await import('../assets/publishConfig.json');
} catch {
  ...
}
```

Reply:
`loadPublishOption` can't be made public because that would require the constructor of `ElectronCapacitorApp` to be async which isn't possible (I can't simply put await in the constructor)